### PR TITLE
Improve S2302 Message: Use NameOf for VB.NET

### LINKED ISSUE #7197: Improve S2302 Message: Use NameOf for VB.NET
S2302 suggests
> Replace the string 'ArgName' with 'nameof(ArgName)'.

Instead, it should suggest this for VB.NET
> Replace the string 'ArgName' with 'NameOf(ArgName)'.

### PR DESCRIPTION
**Pull request by cristian-ambrosini-sonarsource**
Fixes #7197

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CSharp/Rules/NameOfShouldBeUsed.cs [hunk 1]
```diff
@@ -33,6 +33,8 @@ public sealed class NameOfShouldBeUsed : NameOfShouldBeUsedBase<BaseMethodDeclar
 
         protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
 
+        protected override string NameOf => "nameof";
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/NameOfShouldBeUsed.cs:36]: Doesn't need an empty line before

File: analyzers/src/SonarAnalyzer.VisualBasic/Rules/NameOfShouldBeUsed.cs [hunk 2]
```diff
@@ -31,6 +31,8 @@ public sealed class NameOfShouldBeUsed : NameOfShouldBeUsedBase<MethodBlockBaseS
 
         protected override ILanguageFacade<SyntaxKind> Language => VisualBasicFacade.Instance;
 
+        protected override string NameOf => "NameOf";
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.VisualBasic/Rules/NameOfShouldBeUsed.cs:34]: Doesn't need empty line before

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

**Review by gregory-paidis-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**